spec/index.bs

```diff
@@ -6792,68 +6781,58 @@ typedef (sequence<double> or GPUColorDict) GPUColor;
 Note: `double` is large enough to precisely hold 32-bit signed/unsigned
 integers and single-precision floats.
 
-<script type=idl>
-dictionary GPUOrigin2DDict {
-    GPUIntegerCoordinate x = 0;
-    GPUIntegerCoordinate y = 0;
-};
-typedef (sequence<GPUIntegerCoordinate> or GPUOrigin2DDict) GPUOrigin2D;
-</script>
+<!--
+It's a bit of a hack to override the usual conversion to sequence.
+But if we did it explicitly (e.g. with a "typecheck" algorithm), it would infect
+every dictionary that contains one of these members, and every method that takes
+any of those dictionaries or this type.
+-->
```

Comment:
As noted in this comment, an alternative to custom-defining these types would be to define them as `sequence<GPUIntegerCoordinate>`, create "typecheck" algorithms, and call those in all the places these types are used, recursively.

This wouldn't actually be that terrible; I'm already expecting we'll write something like this for GPUBindGroupLayoutEntry, to write in all the "default" values that are currently handwavily defined.